shmem.c

```diff
@@ -476,5 +485,5 @@ static size_t gcd(size_t a, size_t b)
 // in the shared memory object
 static size_t get_optimal_object_size(size_t objsize)
 {
-	return (pagesize*objsize)/gcd(pagesize, objsize);
+	return pagesize / gcd(pagesize, objsize);
```

Comment:
Note that I removed the multiplication by `objsize` here to get the `number of struct elements` to be allocated instead of the number of bytes (`(number of struct elements)*(size of each element in bytes)`). We multiply this number by `sizeof ...` where it is necessary to get the number in Bytes.